### ISSUE TITLE
typing_extensions 4.6.0 breaks phantom types

### ISSUE THREAD

**Issue by diabolo-dan** [antonagestam/phantom-types#277]
```[...]
    from phantom.predicates import numeric
  File "/usr/local/lib/python3.10/site-packages/phantom/predicates/numeric.py", line 3, in <module>
    from phantom._utils.types import SupportsGe
  File "/usr/local/lib/python3.10/site-packages/phantom/_utils/types.py", line 18, in <module>
    class SupportsLt(
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

Seeing this error with the new version of typing_extensions, which was released this morning (4.6.0).  Seems like there are some changes incompatible with the assumptions phantom-types is making.

**Comment by antonagestam** [comment]
Thank you for reporting this, I'll try to look into this soon, but I'm a bit thin on bandwidth at the moment.

**Comment by antonagestam** [comment]
I failed reproducing this and couldn't figure out why. Turns out this is already "accidentally" fixed on `main` 😅

**Comment by antonagestam** [comment]
I just published 2.0.0 with the fix.

**Comment by diabolo-dan** [comment]
Thanks for looking into this quickly

(And thanks also for maintaining the package.  It's very useful).
